### PR TITLE
D9 porting weekend

### PR DESCRIPTION
**Pull request by KarinG**
Overview
----------------------------------------
It's D9 Porting weekend! I worked through this list:
https://www.drupal.org/project/webform_civicrm/issues/3139039

D7 or D8?
----------------------------------------
PR is to make current D8 version -> compatible with D9

Before
----------------------------------------
Issues: https://www.drupal.org/project/webform_civicrm/issues/3139039

After
----------------------------------------
All deprecated APIs listed in https://www.drupal.org/project/webform_civicrm/issues/3139039 have been addressed.

### CODE REVIEW

**Comment by KarinG** [comment]
@jitendrapurohit - this concludes all deprecated function calls on:
https://dev.acquia.com/drupal9/deprecation_status/errors?name=webform_civicrm%20%2A

I'll try to address the parsing errors in a separate PR as those only surface b/c of how upgrade_status is trying to interpret each PHP Class individually. It looks like a new service in `services.yml`

**Comment by KarinG** [comment]
@jitendrapurohit - just also added 
Replace `module_load_include` with `\Drupal::ModuleHandler()->loadInclude`.

**Comment by KarinG** [comment]
Ok - here's the proof -> "No known errors" !

image

**Comment by bmango** [comment]
I have tested this branch with a submission and in the CiviCRM settings and there are no errors coming up.

File: includes/wf_crm_webform_base.inc [hunk 1]
```diff
@@ -859,10 +861,10 @@ abstract class wf_crm_webform_base {
    * @return int|null Civi file id
    */
   public static function saveDrupalFileToCivi($id) {
-    $file = file_load($id);
+    $file = File::load($id);
     if ($file) {
       $config = CRM_Core_Config::singleton();
-      $path = file_unmanaged_copy($file->getFileUri(), $config->customFileUploadDir);
+      $path = FileSystemInterface::copy($file->getFileUri(), $config->customFileUploadDir);
```
Comment (jitendrapurohit) [inline includes/wf_crm_webform_base.inc]: Drupal - 8.8.6, PHP 7.3.7. With the above line, i see an error like -

>Error: Non-static method Drupal\Core\File\FileSystemInterface::copy() cannot be called statically in wf_crm_webform_base::saveDrupalFileToCivi() (line 867 of /Applications/MAMP/htdocs/cividrupal/web/modules/contrib/webform_civicrm/includes/wf_crm_webform_base.inc)

on a pretty complex webform involving activities with file input. Should we call this statically? From https://www.drupal.org/node/3006851, I see `file_unmanaged_copy()` is replaced by `\Drupal::service('file_system')->copy()`. Can you pls confirm?

File: webform_civicrm.module [hunk 2]
```diff
@@ -28,6 +28,7 @@ define('WEBFORM_CIVICRM_DEFAULT_CONTACT_ID', 1);
  */
 function webform_civicrm_library_info_alter(array &$libraries, $extension) {
   if ($extension === 'webform_civicrm') {
+    \Drupal::getContainer()->get('civicrm')->initialize();
```
Comment (jitendrapurohit) [inline webform_civicrm.module]: non blocker, but probably best to write it as `\Drupal::service('civicrm')->initialize();`?

Comment (KarinG) [reply webform_civicrm.module]: I don't understand what's best here -> I just looked at examples from code that I know MattG has touched: 

image

If technically `\Drupal::service('civicrm')->initialize();` is better -> happy to follow your lead on that.

File: includes/wf_crm_webform_base.inc (hunk 1 above)
Comment (KarinG) [reply includes/wf_crm_webform_base.inc]: Ah ok -> this is why I replaced it with ->` FileSystemInterface::copy `-> 

image

Link to the entire report: https://dev.acquia.com/drupal9/deprecation_status/errors?name=webform_civicrm%20%2A

Comment (KarinG) [reply includes/wf_crm_webform_base.inc]: OK seeing that too -> 
image

Comment (KarinG) [reply includes/wf_crm_webform_base.inc]: Does this -> 
`$path = \Drupal::service('file_system')->copy($file->getFileUri(), $config->customFileUploadDir);`
 work for you?

Comment (jitendrapurohit) [reply includes/wf_crm_webform_base.inc]: yes, that works without any problem 👍

File: webform_civicrm.module (hunk 2 above)
Comment (jitendrapurohit) [reply webform_civicrm.module]: pretty sure both are same technically, service() internally calls the same function `getContainer()->get($id)`, but seems the latter is slightly better to read :).

File: includes/wf_crm_webform_base.inc (hunk 1 above)
Comment (KarinG) [reply includes/wf_crm_webform_base.inc]: Excellent - I just updated that!

File: webform_civicrm.module (hunk 2 above)
Comment (KarinG) [reply webform_civicrm.module]: Ok - happy to go with that!

**Comment by KarinG** [comment]
Thanks for your review @jitendrapurohit - I've made the edits!

**Comment by jitendrapurohit** [comment]
works great 👍 . Thanks for the update @KarinG

**Comment by KarinG** [comment]
@jitendrapurohit - thanks for your thorough review -> we now have a first D9 installable release 👍 -> https://www.drupal.org/project/webform_civicrm/releases/8.x-5.0-alpha3

**Pull request merged**